cmd/warm-backend-azure.go

```diff
@@ -49,150 +59,107 @@ func (az *warmBackendAzure) getDest(object string) string {
+	_, err := pager.NextPage(ctx)
 	if err != nil {
-		return nil, err
-	}
-
-	// Refresh obtains a fresh token
-	err = spt.Refresh()
-	if err != nil {
-		return nil, err
+		return false, azureToObjectError(err, az.Bucket, az.Prefix)
 	}
 
-	tc := azblob.NewTokenCredential(spt.Token().AccessToken, func(tc azblob.TokenCredential) time.Duration {
-		err := spt.Refresh()
-		if err != nil {
-			return 0
-		}
-		// set the new token value
-		tc.SetToken(spt.Token().AccessToken)
-
-		// get the next token before the current one expires
-		nextRenewal := float64(time.Until(spt.Token().Expires())) * 0.8
-		if nextRenewal <= 0 {
-			nextRenewal = float64(time.Second)
-		}
-
-		return time.Duration(nextRenewal)
-	})
-
-	return tc, nil
+	return true, nil
 }
 
-func newWarmBackendAzure(conf madmin.TierAzure, _ string) (*warmBackendAzure, error) {
-	var (
-		credential azblob.Credential
-		err        error
-	)
+type azureConf struct {
+	madmin.TierAzure
+}
 
+func (conf azureConf) Validate() error {
 	switch {
 	case conf.AccountName == "":
-		return nil, errors.New("the account name is required")
+		return errors.New("the account name is required")
 	case conf.AccountKey != "" && (conf.SPAuth.TenantID != "" || conf.SPAuth.ClientID != "" || conf.SPAuth.ClientSecret != ""):
-		return nil, errors.New("multiple authentication mechanisms are provided")
+		return errors.New("multiple authentication mechanisms are provided")
 	case conf.AccountKey == "" && (conf.SPAuth.TenantID == "" || conf.SPAuth.ClientID == "" || conf.SPAuth.ClientSecret == ""):
-		return nil, errors.New("no authentication mechanism was provided")
+		return errors.New("no authentication mechanism was provided")
 	}
 
 	if conf.Bucket == "" {
-		return nil, errors.New("no bucket name was provided")
+		return errors.New("no bucket name was provided")
 	}
 
-	if conf.IsSPEnabled() {
-		credential, err = newCredentialFromSP(conf)
-	} else {
-		credential, err = azblob.NewSharedKeyCredential(conf.AccountName, conf.AccountKey)
-	}
-	if err != nil {
-		if _, ok := err.(base64.CorruptInputError); ok {
-			return nil, errors.New("invalid Azure credentials")
-		}
+	return nil
+}
+
+func (conf azureConf) NewClient() (clnt *azblob.Client, clntErr error) {
+	if err := conf.Validate(); err != nil {
 		return nil, err
 	}
-	p := azblob.NewPipeline(credential, azblob.PipelineOptions{})
-	var u *url.URL
-	if conf.Endpoint != "" {
-		u, err = url.Parse(conf.Endpoint)
-		if err != nil {
-			return nil, err
-		}
-	} else {
-		u, err = url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net", conf.AccountName))
+
+	ep := conf.Endpoint
+	if ep == "" {
+		ep = fmt.Sprintf("https://%s.blob.core.windows.net", conf.AccountName)
+	}
+
+	if conf.IsSPEnabled() {
+		credential, err := azidentity.NewClientSecretCredential(conf.SPAuth.TenantID, conf.SPAuth.ClientID, conf.SPAuth.ClientSecret, &azidentity.ClientSecretCredentialOptions{})
```

Comment:
You are allowed to pass `nil` if you don't set any option. This works too...

Reply:
👍🏼